src/iqm/qiskit_iqm/iqm_provider.py

```diff
@@ -241,119 +241,148 @@ def serialize_circuit(self, circuit: QuantumCircuit) -> Circuit:
+        elif instruction.name == 'measure':
+            if len(circuit_instruction.clbits) != 1:
+                raise ValueError(
+                    f'Unexpected: measurement instruction {circuit_instruction} uses multiple classical bits.'
+                )
+            clbit = circuit_instruction.clbits[0]  # always a single-qubit measurement
+            mk = str(MeasurementKey.from_clbit(clbit, circuit))
+            native_inst = Instruction(name='measure', qubits=qubit_names, args={'key': mk})
+            clbit_to_measure[clbit] = native_inst
+        elif instruction.name == 'reset':
+            # implemented using a measure instruction to measure the qubits, and
+            # one cc_prx per qubit to conditionally flip it to |0>
+            feedback_key = '_reset'
+            instructions.append(
+                Instruction(
+                    name='measure',
+                    qubits=qubit_names,
+                    args={
+                        # HACK to get something unique, remove when key can be omitted
+                        'key': f'_reset_{len(instructions)}',
+                        'feedback_key': feedback_key,
+                    },
+                )
+            )
+            for q in qubit_names:
+                physical_qubit_name = index_to_qubit_mapping[int(q)]
                 instructions.append(
                     Instruction(
-                        name='measure',
-                        qubits=qubit_names,
+                        name='cc_prx',
+                        qubits=[q],
                         args={
-                            # HACK to get something unique, remove when key can be omitted
-                            'key': f'_reset_{len(instructions)}',
+                            'angle_t': 0.5,
+                            'phase_t': 0.0,
                             'feedback_key': feedback_key,
+                            'feedback_qubit': physical_qubit_name,
                         },
                     )
                 )
-                for q in qubit_names:
-                    physical_qubit_name = self._idx_to_qb[int(q)]
-                    instructions.append(
-                        Instruction(
-                            name='cc_prx',
-                            qubits=[q],
-                            args={
-                                'angle_t': 0.5,
-                                'phase_t': 0.0,
-                                'feedback_key': feedback_key,
-                                'feedback_qubit': physical_qubit_name,
-                            },
-                        )
-                    )
-                continue
-            elif instruction.name == 'id':
-                continue
+            continue
+        elif instruction.name == 'id':
+            continue
+        else:
+            if ignore_nonnative_gates:
+                args = {f"param{i}": param for i, param in enumerate(instruction.params)}
```

Comment:
```suggestion
                args = {f"p{i}": param for i, param in enumerate(instruction.params)}
```

Reply:
Also pylint will probably suggest making that "else: if:" into another "elif:"

Reply:
Changed.